### ISSUE TITLE
[Feature Request] Use Page Navigation Api for client side routing

### ISSUE THREAD

**Issue by itssumitrai** [sveltejs/kit#5277]
### Describe the problem

The new page navigation api is here

https://developer.chrome.com/docs/web-platform/navigation-api/

This should be preferred over the History Api for client side router. Plus this would also allow for complicated page transition animations which are not possible or pretty difficult to implement through the navigation hooks Sveltekit already provides.

### Describe the proposed solution

Using Page Navigation Api for client side routing
https://developer.chrome.com/docs/web-platform/navigation-api/

### Importance

nice to have

**Comment by tcc-sejohnson** [comment]
https://caniuse.com/mdn-api_navigation

Seems like it's _here_ (as in, on Chrome), but nowhere else. Sounds like a compatibility nightmare -- we'd have to try to use navigation and fall back to the current state.

**Comment by PH4NTOMiki** [comment]
There is also this https://github.com/WICG/navigation-api/issues/66
@Rich-Harris is part of the discussion. And it could take a while.

**Comment by Rich-Harris** [comment]
yeah, this is something to revisit in a year or two. closing until then, there's nothing we can do

**Comment by MentalGear** [comment]
Seems like the time has come: `navigation-api` has reached baseline @Rich-Harris 
https://web.dev/blog/baseline-navigation-api

**Comment by Rich-Harris** [comment]
Yeah we've talked about it on the team. There are enough SvelteKit users serving older browsers that we don't think we can realistically change it for SvelteKit 3, but we definitely want to use it for SvelteKit 4.

**Comment by MentalGear** [comment]
Thanks for the info, though I wonder why not use `navigation-api` for modern browsers and the existing api as legacy fallback?

**Comment by Rich-Harris** [comment]
Because the behaviour will differ in confusing ways, and it would create a significant maintenance burden
